terms/confusionmatrix_df.tex

```diff
@@ -0,0 +1,199 @@
+    \bottomrule
+  \end{tabular}
+  \caption{CM of a hypothetical binary classifier which predicts whether out-of-sample text objects are spam or not. In this example, 10 spam and 100 non-spam objects are classified correctly, whilst 1 spam and 2 non-spam objects are misclassified.}
+  \label{tab:cm_spam}
+\end{table}
+
+Even though CMs are commonly used to evaluate binary classifiers, they aren't restricted to 2-class models \citep{martin2018speech}. A CM of a multi-class model would show the number of times the classes were predicted correctly and which classes were confused with each other (Table \ref{tab:cm_sweets}).
+
+\begin{table}[ht]
+  \centering
+  \fontfamily{ppl}\selectfont
+  \begin{tabular}{llll}
+    \toprule
+                      & \textit{M\&M's} & \textit{Skittles} & \textit{Smarties} \\
+    \midrule
+    \textit{M\&M's}   & 34              & 3                 & 8  \\
+    \textit{Skittles} & 1               & 28                & 5  \\
+    \textit{Smarties} & 2               & 4                 & 22 \\
+    \bottomrule
+  \end{tabular}
+  \caption{CM of a hypothetical sweets classifier. The main diagonal of the CM shows the number of correct predictions, whilst the remaining elements indicate how many sweets were misclassified.}
+  \label{tab:cm_sweets}
+\end{table}
+
+The CM of the model $h : X \mapsto C$ over the concept $c : X \mapsto C$ using dataset $S \subset X$ is formally defined \citep{cichosz2014data} as a matrix $\Xi$ such that $\Xi_{c,S}(h)[d_1,d_2] = |S_{h=d_1,c=d_2}|$. The CM is constructed by incrementing the element corresponding to the true class \textit{vis-a-vi} the predicted class for each object in the dataset (Algorithm \ref{alg:cm}).
+
+\begin{algorithm}
+  \begin{algorithmic}
+    \State $\Xi \gets 0$
+    \For{$x \in S$}
+      \State $d_1 \gets c(x)$
+      \State $d_2 \gets h(x)$
+      \State $\Xi_{d_1,d_2} \gets \Xi_{d_1,d_2} + 1$
+    \EndFor
+  \end{algorithmic}
+  \caption{The CM is initialised to the zero matrix, and populated by iterating over all the objects $x$ with corresponding true class $d_1$ and predicted class $d_2$ and incrementing the element $(d_1,d_2)$ by 1 for each matching outcome.}
+  \label{alg:cm}
+\end{algorithm}
+
+In binary classification, the CM consists of 2 specially designated classes called the \textit{positive} class and the \textit{negative} class \citep{saito2015precision}. As indicated in Table \ref{tab:cm_binary}, positive outcomes from the true class  which are classified correctly are called \textit{True Positives} (TP), whilst misclassifications are called \textit{False Negatives} (FN). On the other hand, negative true class outcomes which are classified correctly are called \textit{True Negatives} (TN), and misclassifications are called \textit{False Positives} (FP). In natural sciences, FP are called \textit{Type I Errors} and FN are known as \textit{Type II Errors} \citep{fielding1997review}.
+
+\begin{table}[ht]
```

Comment:
In general put some vertical space between tables (or equations) and the rest of the text.